lib/plenario_web/controllers/api/detail_controller.ex

```diff
@@ -11,7 +11,25 @@ defmodule PlenarioWeb.Api.DetailController do
 
     def call(conn, opts) do
       meta = MetaActions.get(conn.params["slug"], with_virtual_points: true)
+
+      if meta == nil do
+        conn
+        |> put_req_header("accept", "application/vnd.api+json")
+        |> Explode.with(404, "Data set not found")
+      end
+
       columns = MetaActions.get_column_names(meta)
+      input_params = Map.keys(conn.params)
+      allowed_params = columns ++ ["page", "page_size", "slug", "bbox"]
+
+      for p <- input_params do
```

Comment:
Erring because of an ultimately unused query param seems like overkill, especially when it requires a database call and loop. Also, `404` is not what I would choose -- maybe a 400..?

Reply:
We make the call for  `CaptureArgs` already, and this addresses the issue for case. I used 404 because column/field not found. but 400 works too as an invalid entry

Reply:
But why error at all? If you add useless garbage to any query, it renders the same. Take Google for example:

- https://www.google.com/search?client=ubuntu&channel=fs&q=400+error&ie=utf-8&oe=utf-8
- https://www.google.com/search?q=400+error
- https://www.google.com/search?q=400+error&farts=stink

All render the _exact same results_ even though they have extra parameters beyond the scope of the query (some obvious, some not so obvious is use).